libfreerdp/codec/dsp.c

```diff
@@ -104,46 +109,95 @@ struct _FREERDP_DSP_CONTEXT
  */
 
 static BOOL freerdp_dsp_resample(FREERDP_DSP_CONTEXT* context,
-                                 const BYTE* src, size_t bytes_per_sample,
-                                 UINT32 schan, UINT32 srate, size_t sframes,
-                                 UINT32 rchan, UINT32 rrate)
+                                 const BYTE* src, size_t size,
+                                 const AUDIO_FORMAT* srcFormat,
+                                 const BYTE** data, size_t* length)
 {
 	BYTE* p;
-	int rframes;
-	int rsize;
-	int i, j;
-	int n1, n2;
-	int sbytes, rbytes;
-	sbytes = bytes_per_sample * schan;
-	rbytes = bytes_per_sample * rchan;
-	rframes = sframes * rrate / srate;
-	rsize = rbytes * rframes;
-
-	if (!Stream_EnsureCapacity(context->resample, rsize + 1024))
+	size_t sframes, rframes;
+	size_t rsize;
+	size_t i, j;
+	size_t sbytes, rbytes;
+	size_t srcBytesPerFrame, dstBytesPerFrame;
+	size_t srcChannels, dstChannels;
+	AUDIO_FORMAT format;
+
+	if (!context || !src || !srcFormat || !data || !length)
+		return FALSE;
```

Comment:
static function should not self protect themselves against invalid arguments.

Reply:
@hardening This function is due for a rewrite anyhow ;)
Resampling is quite broken and the sound will be distorted (only exception: `44100Hz` to `22050Hz` or similar special cases)